risc0/zkp/src/hal/cuda.rs

```diff
@@ -564,23 +564,22 @@ impl<CH: CudaHash> Hal for CudaHal<CH> {
         out: &Self::Buffer<Self::ExtElem>,
     ) {
         let po2 = log2_ceil(coeffs.size() / poly_count);
-        assert_eq!(poly_count * (1 << po2), coeffs.size());
+        let count = 1 << po2;
+        assert_eq!(poly_count * count, coeffs.size());
         let eval_count = which.size();
         assert_eq!(xs.size(), eval_count);
         assert_eq!(out.size(), eval_count);
-        let count = 1 << po2;
 
         let stream = Stream::new(StreamFlags::DEFAULT, None).unwrap();
-        let kernel = self.module.get_function("batch_evaluate_any").unwrap();
-        let params = self.compute_simple_params(eval_count);
+        let kernel = self.module.get_function("multi_poly_eval").unwrap();
+        let params = self.compute_simple_params(out.size() * 256);
```

Comment:
I think this is unsafe if default threads per group is not 256.  In general, probably all the 256 constants in the kernel should be blockdim.x as well…